### DISCUSSION TITLE
Selecting previous/next entry in menus is less responsive since recent overhaul

### DISCUSSION THREAD

**Discussion by xfzv** [tomasklaen/uosc#970] [General]
Is it just me or the recent menus overhaul made the navigation a bit laggy? I'm using latest commit on Linux (https://github.com/tomasklaen/uosc/commit/9455db01a79cf2ac5819cf3469bc391f271f7cd5 as I'm writing this).

There's definitely a slight delay when pressing up/down arrow before the previous/next entry is hovered. It's even more noticeable if I keep pressing up or down arrow in a long list.

This occurs in pretty much every menu (main one, Audio, Video, Subtitles, Keybindings, Playlist, [memo](https://github.com/po5/memo) history...)

I can live with it but after reverting to [5.2.0](https://github.com/tomasklaen/uosc/releases/tag/5.2.0), this lag is completely gone. I haven't tried but I guess I could also revert to https://github.com/tomasklaen/uosc/commit/a0a608d451646f989665b521704df13101aad17a since this all started in the last few days.

Just curious if someone else's also noticing the same behavior.

**Comment by Solarunit** [comment]
I noticed this too in 5.3.1 (I skipped 5.3.0) in Audio and Subtitles menus when selecting items with Up/Down arrows.

**Comment by tomasklaen** [comment]
For all practical purposes it is as responsive as before, we just moved the actual handling of keys from keydown to keyup events, as is standard everywhere, so it feels slower because the action happens when you release the key as opposed to when you press it down.

**Reply by FichteFoll** [reply]
> we just moved the actual handling of keys from keydown to keyup events, as is standard everywhere

Are you sure about that? It's most certainly not like this in any text editor I've used in the last decade and I don't see why it would be. Which applications are you thinking of that operate on keyup?

**Comment by mrfragger** [comment]
no notable difference from 5.20, 5.3.1 and 5.4.0 on Mac M1.  Will try Linux in a bit. But I remember 5.3.1 seemed fine.  Haven't updated to 5.4.0 yet though.

**Comment by tomasklaen** [comment]
Fixed by https://github.com/tomasklaen/uosc/commit/3733f71eeac91872d791bc46e87542034d578acf.

**Reply by xfzv** [reply]
Thanks!
